src/codewhisperer/commands/basicCommands.ts

```diff
@@ -120,37 +122,41 @@ export const showSsoSignIn = Commands.declare('aws.codeWhisperer.sso', () => asy
 })
 
 // Shortcut command to directly connect to Identity Center or prompt start URL entry
-// It can optionally set a customization too.
+// It can optionally set a customization too based on given values to match on
 export const connectWithCustomization = Commands.declare(
     'aws.codeWhisperer.connect',
-    () =>
-        async (
-            startUrl?: string,
-            region?: string,
-            customizationArn?: string,
-            customizationName?: string,
-            customizationDescription?: string
-        ) => {
-            // This command supports two sets of arguments:
-            //  * startUrl and region. If both arguments are provided they will be used, otherwise
-            //    the command prompts for them interactively.
-            //  * customization{Arn, Name, Description}. If at least customizationArn is provided,
-            //    the command selects this customization.
-            if (startUrl && region) {
-                await connectToEnterpriseSso(startUrl, region)
-            } else {
-                await getStartUrl()
-            }
-            if (customizationArn) {
-                const match = getPersistedCustomizations().find(c => c.arn == customizationArn)
-                const customization = {
-                    arn: customizationArn,
-                    name: customizationName ?? match?.name ?? 'unknown',
-                    description: customizationDescription ?? match?.description ?? 'unknown',
-                }
-                await selectCustomization(customization)
-            }
+    () => async (startUrl?: string, region?: string, customizationArn?: string, customizationNamePrefix?: string) => {
+        // This command supports the following arguments:
+        //  * startUrl and region. If both arguments are provided they will be used, otherwise
+        //    the command prompts for them interactively.
+        //  * customizationArn: select customization by ARN. If provided, `customizationNamePrefix` is ignored.
+        //  * customizationNamePrefix: select customization by prefix, if `customizationArn` is `undefined`.
+        if (startUrl && region) {
+            await connectToEnterpriseSso(startUrl, region)
+        } else {
+            await getStartUrl()
         }
+
+        // No customization match information given, exit early.
+        if (!customizationArn && !customizationNamePrefix) {
+            return
+        }
+
+        // If given an ARN, assume a specific customization is desired and find an entry that matches it. Ignores the prefix logic.
+        // Otherwise if only a prefix is given, find an entry that matches it.
+        // Backwards compatible with previous implementation.
+        const match: Customization | undefined = customizationArn
```

Comment:
typescript should be able to infer the type
```suggestion
        const match = customizationArn
```

Reply:
How strongly opposed is the team to adding this?
I found it useful for reference on what is being matched, but I agree that on the compiler/linter side it should be able to infer the type.

Reply:
I'll omit it for now.

Reply:
any LSP-capable editor will show the inferred type